pkg/handlers/handlers.go

```diff
@@ -174,18 +143,24 @@ func (h *Handlers) HandleProfiling(w http.ResponseWriter, r *http.Request) {
 
 			// Launch both profilings in parallel as well as the routine to wait for results
 			go func() {
-
-				client := http.DefaultClient
-				client.Transport = http.DefaultTransport
+				client := &http.Client{
+					Transport: newDefaultHTTPTransport(),
+				}
 				client.Transport.(*http.Transport).TLSClientConfig = &tls.Config{RootCAs: h.CACerts, MinVersion: tls.VersionTLS12}
 				hlog.Info("caCertPool loaded in HTTP client Config")
 				runResultsChan <- h.profileKubelet(uid.String(), client)
 			}()
 
 			go func() {
-				connector := connectors.Connector{}
-				connector.Prepare("curl", []string{"--unix-socket", h.CrioUnixSocket, "http://localhost/debug/pprof/profile", "--output", filepath.Join(h.StorageFolder, "crio-"+uid.String()+".pprof")})
-				runResultsChan <- h.profileCrio(uid.String(), &connector)
+				client := &http.Client{
+					Transport: newDefaultHTTPTransport(),
+				}
+				if h.CrioPreferUnixSocket {
+					client.Transport.(*http.Transport).DialContext = func(ctx context.Context, _, _ string) (net.Conn, error) {
+						return (&net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}).DialContext(ctx, "unix", h.CrioUnixSocket)
+					}
+				}
```

Comment:
I think it is better to move this into `h.profileCrio()`

Reply:
Post that we can re-use the `DialContext` being build within `newDefaultHTTPTransport()`

Reply:
Done.